### PR TITLE
  BinarySearch.java: Add binary search recursive algorithm

### PR DESCRIPTION
**Pull request by sudhirsaharan**
THis adds BinarySearch algorithm which searches and returns the index position or position of a
  particular element in an array , if it is not found then it returns
  index value -1

  Closes https://github.com/NITSkmOS/Algorithms/issues/46

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #{ISSUE_NUMBER}

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: binary_search/Java/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        int n = arr.length; 
+        int x = 10; 
+        int result = ob.binarySearch(arr,0,n-1,x); 
+        if (result == -1) 
+            System.out.println("Element not present"); 
+        else
+            System.out.println("Element found at index " +  
+                                                 result); 
+    } 
+}
```
Comment (sangamcse) [inline binary_search/Java/BinarySearch.java]: Add EOL.

Comment (sudhirsaharan) [reply binary_search/Java/BinarySearch.java]: i'm facing issue in squashing the commits and can you please inform what do you mean by adding EOL .

Comment (sangamcse) [reply binary_search/Java/BinarySearch.java]: End of Line

**Comment by sangamcse** [comment]
Update README also

**Comment by sangamcse** [comment]
@sangamcse rebase

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated rebase failed! Please rebase your pull request manually via the command line.

Reason:
```
Command: git rebase base/master

Exit Code: 128

Cause:

.git/rebase-apply/patch:45: trailing whitespace.
/*Binary Search: 
.git/rebase-apply/patch:47: trailing whitespace.
Search a sorted array by repeatedly dividing the search interval in half. 
.git/rebase-apply/patch:48: trailing whitespace.
Begin with an interval covering the whole array. If the value of the search key is 
.git/rebase-apply/patch:49: trailing whitespace.
less than the item in the middle of the interval, narrow the interval to the lower 
.git/rebase-apply/patch:50: trailing whitespace.
half. Otherwise narrow it to the upper half. Repeatedly check until the value is 
warning: squelched 45 whitespace errors
warning: 50 lines add whitespace errors.
error: Failed to merge in the changes.


```

**Comment by sangamcse** [comment]
Also rebase it manually.
http://api.coala.io/en/latest/Developers/Git_Basics.html#rebasing

**Comment by sangamcse** [comment]
Redundant PR

**Pull request closed without merging**